src/store/account-details/account-details.thunks.ts

```diff
@@ -1,102 +1,132 @@
-import axios from 'axios'
-import { CoordinatorAPI } from '@hermeznetwork/hermezjs'
-import { getPoolTransactions } from '@hermeznetwork/hermezjs/src/tx-pool'
-import { TxType } from '@hermeznetwork/hermezjs/src/enums'
-import { push } from 'connected-react-router'
-import * as accountDetailsActions from './account-details.actions'
-import * as ethereum from '../../utils/ethereum'
-import { RootState } from '../'
-import { AppDispatch } from '../../' 
+import axios from "axios";
+import { CoordinatorAPI } from "@hermeznetwork/hermezjs";
+import { getPoolTransactions } from "@hermeznetwork/hermezjs/src/tx-pool";
+import { TxType } from "@hermeznetwork/hermezjs/src/enums";
+import { push } from "connected-react-router";
+import * as accountDetailsActions from "./account-details.actions";
+import * as ethereum from "../../utils/ethereum";
+import { RootState } from "../";
+import { AppDispatch } from "../../";
 
 // domain
-import { Account, Token, HermezTransaction, Exit } from '../../domain/hermez'
-import { HistoryTransactions, HistoryExits } from '../../persistence'
+import { Account, Token, HermezTransaction, Exit } from "../../domain/hermez";
+import { HistoryTransactions, HistoryExits } from "../../persistence";
 
-let refreshCancelTokenSource = axios.CancelToken.source()
+let refreshCancelTokenSource = axios.CancelToken.source();
 
 /**
  * Fetches the account details for the specified account index
  * @param {string} accountIndex - Account index
  * @returns {void}
  */
-function fetchAccount (accountIndex: Account["accountIndex"]) {
+function fetchAccount(accountIndex: Account["accountIndex"]) {
   return (dispatch: AppDispatch, getState: () => RootState) => {
-    const { global: { wallet } } = getState()
+    const {
+      global: { wallet, pricesTask },
+    } = getState();
 
-    dispatch(accountDetailsActions.loadAccount())
+    dispatch(accountDetailsActions.loadAccount());
 
     return CoordinatorAPI.getAccount(accountIndex)
       .then((account: Account) => {
         if (wallet === undefined || account.bjj !== wallet.publicKeyBase64) {
-          dispatch(push('/'))
+          dispatch(push("/"));
         } else {
-          dispatch(accountDetailsActions.loadAccountSuccess(account))
+          const accountTokenUpdated =
```

Comment:
I would create another `createAccount` function which receives an `Account` from the persistence layer and the `pricesTask` and returns the new `Account` from the domain.